### PR TITLE
Allow the governance to recover ERC20 and ERC721 from TBTCVault 

### PR DESCRIPTION
**Pull request by pdyraga**
~~Depends on #309. Keeping as a draft until #309 is not merged.~~
Refs: https://github.com/keep-network/tbtc-v2/issues/39

ERC20 and ERC721 can be sent mistakenly to `TBTCVault`, two functions allow the
governance to recover them. Most importantly though, the governance can use
`recoverERC20` to recover TBTCv1 tokens sent by `VendingMachine` to `TBTCVault`
after the upgrade.

### CODE REVIEW

File: solidity/contracts/vault/TBTCVault.sol [hunk 1]
```diff
@@ -249,6 +223,64 @@ contract TBTCVault is IVault, Governable {
         upgradeInitiatedTimestamp = 0;
     }
 
+    /// @notice Allows the governance of the TBTCVault to recover any ERC20
+    ///         token sent mistakenly to the TBTC token contract address.
+    /// @param token Address of the recovered ERC20 token contract.
+    /// @param recipient Address the recovered token should be sent to.
```
Comment (tomaszslabon) [inline solidity/contracts/vault/TBTCVault.sol:229]: `@param`'s should not end with a dot.

File: solidity/contracts/vault/TBTCVault.sol [hunk 2]
```diff
@@ -249,6 +223,64 @@ contract TBTCVault is IVault, Governable {
...
+        tbtcToken.recoverERC721(token, recipient, tokenId, data);
+    }
+
+    /// @notice Allows the governance of the TBTCVault to recover any ERC20
+    ///         token sent - mistakenly or not - to the vault address. This
+    ///         function should be used to withdraw TBTC v1 tokens transferred
+    ///         to TBTCVault as a result of VendingMachine > TBTCVault upgrade.
+    /// @param token Address of the recovered ERC20 token contract.
+    /// @param recipient Address the recovered token should be sent to.
+    /// @param amount Recovered amount.
```
Comment (tomaszslabon) [inline solidity/contracts/vault/TBTCVault.sol:260]: @param's should not end with a dot.

File: solidity/contracts/vault/TBTCVault.sol [hunk 3]
```diff
@@ -249,6 +223,64 @@ contract TBTCVault is IVault, Governable {
...
+    ) external onlyGovernance {
+        token.safeTransfer(recipient, amount);
+    }
+
+    /// @notice Allows the governance of the TBTCVault to recover any ERC721
+    ///         token sent mistakenly to the vault address.
+    /// @param token Address of the recovered ERC721 token contract.
+    /// @param recipient Address the recovered token should be sent to.
+    /// @param tokenId Identifier of the recovered token.
+    /// @param data Additional data.
```
Comment (tomaszslabon) [inline solidity/contracts/vault/TBTCVault.sol:274]: @param's should not end with a dot.

File: solidity/contracts/vault/TBTCVault.sol (hunk 1 above)
Comment (pdyraga) [reply solidity/contracts/vault/TBTCVault.sol:229]: It should now 😆 We changed it in https://github.com/keep-network/tbtc-v2/pull/281:

> Each `@param` ends with a dot `.` now. It is a different approach from the one we
used in `@keep-core/ecdsa` and `@keep-core/random-beacon` but it is time to
acknowledge `@param` descriptions in `tbtc-v2` are much longer and they often do
form (multiple) sentences.

**Review by tomaszslabon — APPROVED** [review]

**Comment by pdyraga** [comment]
#309 is merged so I am undrafting this PR.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**